extensions/ql-vscode/src/query-history.ts

```diff
@@ -611,7 +613,29 @@ export class QueryHistoryManager extends DisposableObject {
       await this.refreshTreeView();
     });
 
+    const variantAnalysisStatusUpdateSubscription = this.variantAnalysisManager.onVariantAnalysisStatusUpdated(async (variantAnalysis) => {
+      const items = this.treeDataProvider.allHistory.filter(i => i.t === 'variant-analysis' && i.variantAnalysis.id === variantAnalysis.id);
+      const status = variantAnalysisStatusToQueryStatus(variantAnalysis.status);
+
+      if (items.length > 0) {
+        items.forEach(async (item) => {
+          const variantAnalysisHistoryItem = item as VariantAnalysisHistoryItem;
+          variantAnalysisHistoryItem.status = status;
+          variantAnalysisHistoryItem.failureReason = variantAnalysis.failureReason;
+          variantAnalysisHistoryItem.resultCount = getTotalResultCount(variantAnalysis.scannedRepos);
+          variantAnalysisHistoryItem.variantAnalysis = variantAnalysis;
+          if (status === QueryStatus.Completed) {
+            variantAnalysisHistoryItem.completed = true;
+          }
+          await this.refreshTreeView();
```

Comment:
Thanks for making this work on all items except just one. In normal operation I'd hope there should just be one history item per variant analysis, but this does seem like a nice way of making things work in case that invariant is breached.

However, should we move this call to `refreshTreeView` to be after the loop?

Reply:
Have moved the call. Thanks!